test/asynchronous/test_client.py

```diff
@@ -2020,6 +2024,78 @@ def test_dict_hints_sort(self):
     async def test_dict_hints_create_index(self):
         await self.db.t.create_index({"x": pymongo.ASCENDING})
 
+    @async_client_context.require_connection
```

Comment:
I beleive we can remove the `require_connection`s now that this is in test_client.